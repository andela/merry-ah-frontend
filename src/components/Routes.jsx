import 'babel-polyfill';
import React from "react";
import Home from "./Home";
import NotFound from "./NotFound"
import store from '../store';
import { Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux'
import Auth from "./authentication/Auth";
import Profile from './Profile';

const Routes = () => (
    <Provider store={store}>
    <Switch>
        <Route exact path="/" component={Home}></Route>
        <Route path="/auth" component={Auth}></Route>
        <Route path="/profile" component={Profile}></Route>
        <Route component={NotFound}></Route>
    </Switch>
    </Provider>
);

export default Routes;
