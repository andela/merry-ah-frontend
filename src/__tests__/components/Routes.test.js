import React from 'react';
import { shallow } from 'enzyme';
import Routes from '../../components/Routes';

let wrapper;
beforeEach(() => {
 wrapper = shallow(<Routes />);
})
describe('Routes Component', () => {
  it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
  })
})
