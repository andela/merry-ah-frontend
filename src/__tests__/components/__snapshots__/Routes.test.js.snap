// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component should match snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/art/:category"
  />
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    path="/login"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    path="/profile"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
